### PR TITLE
feat(search): small Search Dialog improvements

### PR DESCRIPTION
**Pull request by vdegenne**
- Add "no results" feedback
- Focus search input when search dialog opens (FF, Safari, ...)

### CODE REVIEW

File: packages/lit-dev-content/src/components/litdev-search-modal.ts [hunk 1]
```diff
@@ -8,7 +8,7 @@ import {css, html, LitElement, PropertyValues} from 'lit';
 import {customElement, property, query} from 'lit/decorators.js';
 import './lazy-svg.js';
 import './litdev-search.js';
-import type {LitDevSearch} from './litdev-search.js';
+import {LitDevSearch} from './litdev-search.js';
```
Comment (e111077) [inline packages/lit-dev-content/src/components/litdev-search-modal.ts:11]: This was right before, please revert this line

File: packages/lit-dev-content/src/components/litdev-search.ts [hunk 2]
```diff
@@ -224,9 +224,17 @@ export class LitDevSearch extends LitElement {
...
+        ? html`
+            <ul
+              id="items"
+              role="listbox"
+              class=${classMap({expanded: items.length})}
+            >
+              ${this._renderGroups()}
+            </ul>
+          `
+        : html`<div id="no-items">no results</div>`}
```
Comment (e111077) [inline packages/lit-dev-content/src/components/litdev-search.ts:237]: Can you please link to a new bug report on this repo with something prefilled like:

```
Subject: No search results for `${this.searchText}`

Content: <!-- What type of content did you expect to see on lit.dev and explain why it should be on lit.dev -->
```

Here's a nifty article on how to generate a URL for prefilled github issues:

https://eric.blog/2016/01/08/prefilling-github-issues/

**Comment by e111077** [comment]
deployed to: https://pr1067-19b7994---lit-dev-playground-5ftespv5na-uc.a.run.app

File: packages/lit-dev-content/src/components/litdev-search.ts [hunk 3]
```diff
@@ -224,9 +224,17 @@ export class LitDevSearch extends LitElement {
           Cancel
         </button>
       </div>
-      <ul id="items" role="listbox" class=${classMap({expanded: items.length})}>
-        ${this._renderGroups()}
-      </ul>
+      ${this._searchText == '' || items.length > 0
```
Comment (e111077) [inline packages/lit-dev-content/src/components/litdev-search.ts:227]: we do not search unless `this._searchText.length > 1` please update

**Review by e111077 — CHANGES_REQUESTED** [review]
Thanks, I like this!

**Comment by e111077** [comment]
deployed to:
https://pr1067-ee3b1a4---lit-dev-5ftespv5na-uc.a.run.app/

File: packages/lit-dev-content/src/components/litdev-search.ts (hunk 2 above)
Comment (e111077) [reply packages/lit-dev-content/src/components/litdev-search.ts]: Pseudocode here could be:


```ts
html`<a href=${this.getGithubIssueUrl(this.searchText)} target="blank">No results</a>`
...
getGithubIssueUrl(searchText: string) {
  const githubIssueUrl = new URL('https://github.com/lit/lit.dev/issues/new');
  githubIssueUrl.searchParams
      .append('title', `[docs] No search results for \`${searchText}\``);
  githubIssueUrl.searchParams
      .append('body', `<!-- What type of content did you expect to see on lit.dev and explain why it should be on lit.dev -->`);
  return githubIssueUrl.href;
}
```

Comment (vdegenne) [reply packages/lit-dev-content/src/components/litdev-search.ts]: Is it ok if I write this function in `url-helper.ts` in `util` ?

**Comment by e111077** [comment]
Deployed to:

https://pr1067-d1f08d2---lit-dev-5ftespv5na-uc.a.run.app/

**Review by e111077 — APPROVED** [review]
Stellar work! It should deploy to prod in like 10ish minutes

**Pull request merged**